src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/DefaultGetDraftViewOperation.java

```diff
@@ -0,0 +1,102 @@
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.data.definition.UIDefinitionRepository;
+import uk.gov.hmcts.ccd.data.draft.DefaultDraftGateway;
+import uk.gov.hmcts.ccd.data.draft.DraftGateway;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CaseView;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CaseViewEvent;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CaseViewTrigger;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CaseViewType;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTabCollection;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseType;
+import uk.gov.hmcts.ccd.domain.model.draft.CaseDraft;
+import uk.gov.hmcts.ccd.domain.model.draft.DraftResponse;
+import uk.gov.hmcts.ccd.domain.service.common.CaseTypeService;
+import uk.gov.hmcts.ccd.domain.service.common.UIDService;
+import uk.gov.hmcts.ccd.domain.service.getcase.CreatorGetCaseOperation;
+import uk.gov.hmcts.ccd.domain.service.getcase.GetCaseOperation;
+
+import static uk.gov.hmcts.ccd.domain.model.aggregated.CaseViewTriggerBuilder.aCaseViewTrigger;
+import static uk.gov.hmcts.ccd.domain.model.definition.CaseDetailsBuilder.aCaseDetails;
+
+@Service
+@Qualifier(DefaultGetDraftViewOperation.QUALIFIER)
+public class DefaultGetDraftViewOperation extends AbstractDefaultGetCaseViewOperation implements GetCaseViewOperation {
+
+    public static final String QUALIFIER = "defaultDraft";
+    static final String DRAFT_ID = "DRAFT%s";
+    private static final CaseViewTrigger DELETE_TRIGGER = aCaseViewTrigger()
+        .withName("Delete")
+        .withDescription("Delete draft")
+        .withOrder(2)
+        .build();
+    private static final String RESUME = "Resume";
+
+    private final DraftGateway draftGateway;
+
+    @Autowired
+    public DefaultGetDraftViewOperation(@Qualifier(CreatorGetCaseOperation.QUALIFIER) final GetCaseOperation getCaseOperation,
+                                        final UIDefinitionRepository uiDefinitionRepository,
+                                        final CaseTypeService caseTypeService,
+                                        final UIDService uidService,
+                                        @Qualifier(DefaultDraftGateway.QUALIFIER) final DraftGateway draftGateway) {
+        super(getCaseOperation, uiDefinitionRepository, caseTypeService, uidService);
+        this.draftGateway = draftGateway;
+    }
+
+    @Override
+    public CaseView execute(String jurisdictionId, String caseTypeId, String draftId) {
+
+        final CaseType caseType = getCaseType(jurisdictionId, caseTypeId);
+        final DraftResponse draftResponse = draftGateway.get(draftId);
+
+        final CaseDetails caseDetails = buildCaseDetailsFromDraft(draftResponse);
+        final CaseViewTrigger resumeTrigger = buildResumeTriggerFromDraft(draftResponse);
```

Comment:
could we explicitly model a resume trigger with a subclass of CaseViewTrigger called ResumeTrigger? It could provide a constructor which accepts id and description. 
This way we could also remove the builder

Reply:
I would avoid extracting this as a new class as there is no difference in content. I think the method name `buildResumeTriggerFromDraft` as well as the variable name `resumeTrigger` should be enough to understand it here.